eclipse-theia/theia
# Add validations to Monaco

#### What it does


Fixes #11241 by removing the buggy `ValidatedPreferenceProxy` implementation in favor of adding validations for a few key preferences to Monaco's own validation system for editor preferences.

#### How to test


1. Open a workspace and open a text file in that workspace.
2. In that workspace, set e.g. `editor.fontSize` to some non-default value.
3. Observe that the new setting takes effect in your open file.
4. In user scope, set the same preference to some other non-default value.
5. Observe that that new setting does _not_ take effect in your open file because it is overridden in a higher-priority scope.

---

1. Also confirm that the testing steps for #10607 and #10965 still go as expected.


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
